acestep/llm_inference.py

```
@@ -125,13 +125,23 @@ def _clear_accelerator_cache(self) -> None:
     def unload(self) -> None:
         """Release LM weights/tokenizer and clear caches to free memory."""
         try:
-            if self.llm_backend == "vllm":
+            if self.llm_backend == "vllm" and getattr(self, "llm", None) is not None:
                 try:
-                    if hasattr(self.llm, "reset"):
-                        self.llm.reset()
+                    if hasattr(self.llm, "exit"):
+                        self.llm.exit()  # full teardown: weights/KV cache/CUDA graphs/subprocesses
                 except Exception:
                     pass
```

Ruff (0.16.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)